### ISSUE TITLE
2.0.1 Can't change mode via flows

### ISSUE THREAD

**Issue by lenwar** [caseda/org.eurotronic#20]
Using Spirits.

When I try to change the heating type via a flow, it marks it as  'done' but in the end it doesn't do anything. It shoots no error. It just doesn't change the setting.

Changing it 'manually' works properly.

**Comment by caseda** [comment]
I'll inspect what's up, thanks for reporting

**Comment by lenwar** [comment]
Hi,
It appears to work now. But it doesn't show on the device directly (caching?). It takes some time, minuten to show.

If I make a flow:
When flow is started
And Spirit is not set to Economic
Then set Spirit to Economic

It works

**Comment by lenwar** [comment]
Hi,

I made a mistake in thinking this was resolved.
During my testing for [Issue 17](
https://github.com/caseda/org.eurotronic/issues/17) I found the following, so unlike what I found above:

The ‘current’ setting:
Valve is in Comfortable mode
Valve comfortable temperature is 20 degrees
The screen on the valve shows 20 degrees

The flow:
1 When flow is started
2 And Spirit is NOT set to Economic
3 Then set Spirit to Economic

I start the flow:
It passes 1 and 2 and executes 3 succesfully (it marks it as ‘done’ - checkmark)
The device view still shows the valve setting as ‘Comfortable’

I start the flow again:
It passes 1 and fails on 2
The device view still shows the valve setting as ‘Comfortable’

I restarted the Homey app to confirm:
The device view still shows the valve setting as ‘Comfortable’

I start the flow yet again:
It passes 1 and 2 and executes 3 succesfully (it marks it as ‘done’ - checkmark)
The device view still shows the valve setting as ‘Comfortable’


My conclusions:
The flow runs properly at checking if the valve is no in economic mode and tries to set it to economic mode
When running the flow again, it detects that the valve is set to economic mode due to the first run, yet it isn’t so.
After restarting the app and rerunning the flow it properly detects that the valve is not economic mode and can rerun it though in the end it doesn’t appear to do anything.
